number_theory/utility.h

```diff
@@ -38,11 +38,9 @@ template <class T>
 constexpr std::size_t bit_length(T number) {
   static_assert(std::numeric_limits<T>::is_integer,
                 "bit_length argument must be an integer.");
-  using Unsigned_T = std::make_unsigned_t<T>;
-  Unsigned_T abs_number = std::abs(number);
-  return std::numeric_limits<Unsigned_T>::digits - std::countl_zero(abs_number);
+  return std::bit_width(number);
```

Comment:
`std::bit_width` is only available for unsigned integers, so we need to use `std::abs(number)` and cast to unsigned integer as before.

Reply:
In Python, negative numbers are also able to call `int.bit_length`.